apps/prairielearn/src/tests/serverJobs.test.ts

```diff
@@ -73,4 +73,30 @@ describe('server-jobs', () => {
     assert.equal(job.status, 'Error');
     assert.match(stripAnsi(job.output), /^testing info\nError: failing job\n\s+at/);
   });
+
+  it('fails the job when fail() is called', async () => {
+    const serverJob = await createServerJob({
+      type: 'test',
+      description: 'test job sequence',
+    });
+
+    await assert.isRejected(
+      serverJob.execute(async (job) => {
```

Comment:
For coverage/completeness would it be a good idea to also test `fail` in a background job? In particular that the function returns without an exception but that the output includes the error message?

Reply:
Sure, I can add a test for that.

Reply:
Done in 9ff1095.